onadata/libs/utils/export_tools.py

```diff
@@ -163,9 +163,17 @@ def generate_export(export_type, xform, export_id=None, options=None,
     if options.get("dataview_pk"):
         dataview = DataView.objects.get(pk=options.get("dataview_pk"))
         records = dataview.query_data(dataview, all_data=True)
+        total_records = dataview.query_data(dataview,
+                                            count=True)[0].get('count')
     else:
         records = query_data(xform, query=filter_query, start=start, end=end)
 
+        if filter_query:
+            total_records = query_data(xform, query=filter_query, start=start,
+                                       end=end, count=True)[0].get('count')
```

Comment:
This makes a huge assumption that we will always receive a list of at least size one.